src/wallet/wallettool.cpp

```diff
@@ -29,10 +29,7 @@ static void WalletCreate(CWallet* wallet_instance)
     // generate a new HD seed
     wallet_instance->SetupLegacyScriptPubKeyMan();
     auto spk_man = wallet_instance->GetLegacyScriptPubKeyMan();
-    // NOTE: drop this condition after removing option to create non-HD wallets
-    if (spk_man->IsHDEnabled()) {
-        spk_man->GenerateNewHDChain(/*secureMnemonic=*/"", /*secureMnemonicPassphrase=*/"");
-    }
+    spk_man->GenerateNewHDChain(/*secureMnemonic=*/"", /*secureMnemonicPassphrase=*/"");
```

Comment:
we still allow non-hd correct?

Reply:
indeed, probably it would fail for non-hd.
`GenerateNewHDChain` has never been called before, not it calls always. I will prepare a fix

Reply:
see https://github.com/dashpay/dash/commit/31ffb78cedb8cf167a7fbc9240d947e1814590a1